### ISSUE TITLE
gitkraken: commit graph not rendered

### ISSUE THREAD

**Issue by sheepforce** [NixOS/nixpkgs#267434]
### Describe the bug
Since the gitkraken 9.5.1 -> 9.9.2 update (https://github.com/NixOS/nixpkgs/commit/a4c90e09f28b578d48cee150dceb3b94f5b9b554), gitkraken is not showing the commit graph anymore, making it effectively unusable.

### Steps To Reproduce
1. `NIXPKGS_ALLOW_UNFREE=1 nix run --impure nixpkgs/a4c90e09f28b578d48cee150dceb3b94f5b9b554#gitkraken -- --disable-gpu` (does not render commit graph)
2. `NIXPKGS_ALLOW_UNFREE=1 nix run --impure nixpkgs/e6910fd16e77f1f5f15cd5e22d725d6afd6ef5ee#gitkraken -- --disable-gpu` (works, renders commit graph)

### Expected behavior
The context graph should be visible.

### Screenshots
after a4c90e09f28b578d48cee150dceb3b94f5b9b554 
![GitkrakenNonWorking](https://github.com/NixOS/nixpkgs/assets/16844216/6120e776-fc92-4aba-ab04-89b3a8fe3e6f)

before e6910fd16e77f1f5f15cd5e22d725d6afd6ef5ee 
![GitkrakenWorking](https://github.com/NixOS/nixpkgs/assets/16844216/528a76bc-5821-4aa5-bc95-87e78260a311)

### Additional context
I'm seeing the EGL erros described in #158947 without the `--disable-gpu` flag. However, I'm also seeing them in the working versions and they disappear when running with `--disable-gpu`, yet it still does not show the commit graph.

### Notify maintainers
@xNWDD 
@evanjs
@arkivm

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.10, NixOS, 23.11 (Tapir), 23.11.20231104.85f1ba3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Ciflire** [comment]
I'm having the same issue

**Comment by nicolas-goudry** [comment]
I had the same issue. The repositories on which it happened were corrupted.

Try to run `git fsck` on the repositories where the commit graph doesn’t render.

Only way I found to fix this was to get a fresh clone of the repository elsewhere and copy my current changes in the new repository location.

If you have a better way to fix it, I’m all ears, since this is still happening on some of my repositories (but I don’t need to work on them rn so I didn’t bother doing that yet).

**Comment by sheepforce** [comment]
I'm afraid it's hitting all repos, even freshly cloned ones and also after running `git fsck`. This doesn't seem to be the problem here :grimacing:

**Comment by nicolas-goudry** [comment]
Frack, that’s unexpected… It works fine on my machine. Though I’m locally using the updated Gitkraken 9.10.0, it used to also work with Gitkraken 9.9.2.

Could you maybe try with the [next coming update](https://github.com/NixOS/nixpkgs/pull/272139) ?

**Comment by sheepforce** [comment]
> Could you maybe try with the https://github.com/NixOS/nixpkgs/pull/272139 ?

Sure! :) 

```bash
$ NIXPKGS_ALLOW_UNFREE=1 nix run --impure github:nicolas-goudry/nixpkgs/gitkraken-update#gitkraken
[179043:1205/134101.849707:ERROR:angle_platform_impl.cc(43)] Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
ERR: Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179043:1205/134101.850031:ERROR:gl_display.cc(508)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179043:1205/134101.850088:ERROR:gl_display.cc(920)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
[179043:1205/134101.850529:ERROR:angle_platform_impl.cc(43)] Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
ERR: Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179043:1205/134101.850621:ERROR:gl_display.cc(508)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179043:1205/134101.850662:ERROR:gl_display.cc(920)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[179043:1205/134101.850721:ERROR:gl_ozone_egl.cc(23)] GLDisplayEGL::Initialize failed.
[179043:1205/134101.876991:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization
[179244:1205/134110.128923:ERROR:angle_platform_impl.cc(43)] Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
ERR: Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179244:1205/134110.129529:ERROR:gl_display.cc(508)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179244:1205/134110.129599:ERROR:gl_display.cc(920)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
[179244:1205/134110.129943:ERROR:angle_platform_impl.cc(43)] Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
ERR: Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179244:1205/134110.130035:ERROR:gl_display.cc(508)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
[179244:1205/134110.130074:ERROR:gl_display.cc(920)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[179244:1205/134110.130110:ERROR:gl_ozone_egl.cc(23)] GLDisplayEGL::Initialize failed.
[179244:1205/134110.156738:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization
ldd: warning: you do not have execution permission for `/nix/store/68148vb5vhv3141vdbhdih1y9q4dwgp8-gitkraken-9.10.0/share/gitkraken/resources/app.asar.unpacked/node_modules/@axosoft/nodegit/build/Release/nodegit-ubuntu-18.node'
(node:168620) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `gitkraken --trace-deprecation ...` to show where the warning was created)
(node:168620) [DEP0106] DeprecationWarning: crypto.createDecipher is deprecated.
[179300:1205/134111.600397:ERROR:angle_platform_impl.cc(43)] Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory
```

Same issue, still even when cloning a fresh repo. The most fishy looking message seems to be `ERR: Display.cpp:1004 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: libGL.so.1: cannot open shared object file: No such file or directory`. Maybe some patchelf forcing libGL into the rpath could help? I'm just speculating, though :smile:

**Comment by nicolas-goudry** [comment]
Shoot. Did you try running it with the silly debug messages ?

```
--d SILLY
```

**Comment by sheepforce** [comment]
Not too much difference, I'm afraid:
```
phillip@rlyeh:~/Dokumente/Git/nixpkgs$ NIXPKGS_ALLOW_UNFREE=1 nix run --impure .#gitkraken -- -d SILLY |& tee gitkraken.log
warning: Git tree '/home/phillip/Dokumente/Git/nixpkgs' is dirty
ldd: warning: you do not have execution permission for `/nix/store/dk664njk5zjx0b3nwvgzidyrw7a90mz7-gitkraken-9.9.2/share/gitkraken/resources/app.asar.unpacked/node_modules/@axosoft/nodegit/build/Release/nodegit-ubuntu-18.node'
(node:220092) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `gitkraken --trace-deprecation ...` to show where the warning was created)
(node:220092) [DEP0106] DeprecationWarning: crypto.createDecipher is deprecated.
state:  unavailable
ldd: warning: you do not have execution permission for `/nix/store/dk664njk5zjx0b3nwvgzidyrw7a90mz7-gitkraken-9.9.2/share/gitkraken/resources/app.asar.unpacked/node_modules/@axosoft/nodegit/build/Release/nodegit-ubuntu-18.node'
ldd: warning: you do not have execution permission for `/nix/store/dk664njk5zjx0b3nwvgzidyrw7a90mz7-gitkraken-9.9.2/share/gitkraken/resources/app.asar.unpacked/node_modules/@axosoft/nodegit/build/Release/nodegit-ubuntu-18.node'
(node:222721) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `exe --trace-deprecation ...` to show where the warning was created)
state:  checking
(node:222721) [DEP0106] DeprecationWarning: crypto.createDecipher is deprecated.
state:  available
```
(this is current version 9.9.2 from master with libGL added to the rpath, which reduces the noise).

I've tried the flatpack version 9.10.0 which works flawlessly.

**Comment by sheepforce** [comment]
I was just thinking that it could be some NVidia issue but it is apparently not. Doesn't work on my desktop with proprietary nvidia drivers under X11 but also not on my laptop with a Ryzen APU and mesa drivers (just to exclude another option).

**Comment by sheepforce** [comment]
Crazy, that actually worked! "This opt-out is temporary. Please contact us with any issues you have" <- done, and I'm actually a paying customer, so let's see :)

**Comment by nicolas-goudry** [comment]
From what I understand, this has to do with hardware acceleration? Not sure why this hasn’t been added earlier…

Since I can’t reproduce the same logs as yours, would you be kind enough to test it locally on your machine so I can include this change in my PR?

**Comment by nicolas-goudry** [comment]
@sheepforce I have just been reminded that I had to disable the bundled git binary to make the graph work. Could you try to:
1. Enable back the new commit graph (`Preferences > Experimental > New Commit Graph`)
2. Set the « Git executable » to the non-bundled one (`Preferences > Experimental > Git Executable`)

Make sure that you don’t see the errors related to `libGL` with these settings. If you still see them, I believe a fix could be applied but it should be on its own PR IMO.

**Comment by nicolas-goudry** [comment]
Actually I will patch 9.10.0 to include the `git` binary from nixpkgs so we don’t have to perform this workaround.

I’m currently struggling to fix the UI so that it shows the correct version being used, but it’s a detail. I might fix this in a separate PR. When 9.10.0 will be released, tweaking this setting won’t have any effect, since it will always use the `git` binary from nix no matter what you choose.

**Comment by flexiondotorg** [comment]
I've filled an issue with GitGraken too.

**Comment by flexiondotorg** [comment]
@nicolas-goudry  Your nifty git patch is still the most robust solution right now :+1: So I'd keep it in your current PR for GitKraken 9.10.0.

Perhaps we keep this issue open to try and find a solution that wrangles the bundled git executables so they work?

- I noticed the bundled `git` and `libexec/git-core` in GitKraken requires `zlib`, so I've included that in my local build and corrected the rpath. Even so, the commit graph fails to render.

- The bundled NodeGit that `gitkraken` invokes is linked against OpenSSL 1.1.1, which is now marked as insecure in nixpkgs 23.11, so not something we should ship. Not sure how to take that.

Here's my gitkraken config which is derived from your https://github.com/NixOS/nixpkgs/pull/272139, adding `zlib`, `libGL` and correcting the rpath for the assorted git binaries.

- https://github.com/wimpysworld/nix-config/blob/main/pkgs/gitkraken.nix
